metricflow/dataflow/dataflow_plan.py

```diff
@@ -454,11 +454,38 @@ class SemiAdditiveJoinNode(Generic[SourceDataSetT], BaseOutput[SourceDataSetT]):
     | date       | account_balance | user |                             | date       | account_balance | user |
     |:-----------|-----------------|-----:|     identifier_specs:       |:-----------|-----------------|-----:|
     | 2019-12-31 |            1000 |    u1|       - user                | 2020-01-03 |            2000 |    u1|
-    | 2020-01-03 |            2000 |    u1| ->  time_dimension_spec: -> | 2020-01-11              1500 |    u2|
+    | 2020-01-03 |            2000 |    u1| ->  time_dimension_spec: -> | 2020-01-12              1500 |    u2|
     | 2020-01-09 |            3000 |    u2|       - date                | 2020-01-12 |            1000 |    u3|
-    | 2020-01-11 |            1500 |    u2|     agg_by_function:
+    | 2020-01-12 |            1500 |    u2|     agg_by_function:
     | 2020-01-12 |            1000 |    u3|       - MAX
 
+
+    Similarly, if we don't provide any identifier_specs, it would end up performing the aggregation filter only on the
+    time_dimension_spec without any grouping by any identifiers.
+
+    Data transformation example,
+    | date       | account_balance | user |                             | date       | account_balance |
+    |:-----------|-----------------|-----:|     identifier_specs:       |:-----------|----------------:|
+    | 2019-12-31 |            1000 |    u1|                             | 2020-01-12 |            2500 |
+    | 2020-01-03 |            2000 |    u1| ->  time_dimension_spec: ->
+    | 2020-01-09 |            3000 |    u2|       - date
+    | 2020-01-12 |            1500 |    u2|     agg_by_function:
+    | 2020-01-12 |            1000 |    u3|       - MAX
+
+
+    Additionally, we can aggregate against 'windows' of a dataset. For example, if we group by a non_additive time dimension,
+    we perform the MIN/MAX filtering on the granularity window provided in that group by instead of the full time range.
+
+    Data transformation example,
+    | date       | account_balance | user |                                  | date       | account_balance |
+    |:-----------|-----------------|-----:|     identifier_specs:            |:-----------|----------------:|
+    | 2019-12-31 |            1500 |    u1|     time_dimension_spec:         | 2019-12-31 |            1500 |
+    | 2020-01-03 |            2000 |    u1| ->    - date                  -> | 2020-01-07 |            3000 |
+    | 2020-01-09 |            3000 |    u2|     agg_by_function:             | 2020-01-14 |            3250 |
+    | 2020-01-12 |            1500 |    u2|       - MIN
+    | 2020-01-14 |            1250 |    u3|     queried_time_dimension_spec:
+    | 2020-01-14 |            2000 |    u2|       - date__week
```

Comment:
Never mind, I think I was getting confused by the identifier in the data itself.

The contract for semi-additive measures is that we can aggregate *within* values for the non-additive dimension, but not *across* them, and this is exactly what happens here. While it might be odd for somebody to ignore the `user` entry in this case, it's not wrong - in fact, if they really don't care about the `user` under normal circumstances (or if they're never allowed to have user-level aggregations in their reporting pipeline) this is exactly what they'd want.